File: 02 tests/ex8_lists.js
// run with: deno run ex8_lists.js | node ex8_lists.js

/* ex8_lists.js

List comprehension vs. Array mapping

:author:  wolf
:created: 2025.05.22
*/

const iterations = 3;
const timing_results = [];

// define the test function
// --- function to be timed-- -

function jsArrayMap(n) {
    return Array.from({ length: n }, (_, i) => i * i);
}


for (let i = 0; i < iterations; i++) {
    const start = performance.now();

    // call the test function
    jsArrayMap(10**7);

    const end = performance.now();

    timing_results.push(end - start);
}


console.log("Elapsed time: ",
    timing_results.reduce((a, b) => a + b) / iterations / 1000, "seconds");

File: 02 tests/ex8_lists.py
#!/usr/bin/env python3

""" ex8_lists.py

List comprehension vs. Array mapping

:author:	wolf
:created:	2025.05.22
"""

import time

iterations = 3
timing_results = []

# define the test function
# --- function to be timed ---


def python_list_comprehension(n):
    return [i * i for i in range(n)]


for i in range(iterations):
    start = time.perf_counter()

    # call the test function
    python_list_comprehension(10**7)

    end = time.perf_counter()

    timing_results.append(end - start)


print(f"Elapsed time:  {sum(timing_results)/iterations:.16f} seconds")
